src/js/_enqueues/wp/heartbeat.js

```diff
@@ -733,35 +734,28 @@
 				oldInterval = settings.tempInterval ? settings.tempInterval : settings.mainInterval;
 
 			if ( speed ) {
-				switch ( speed ) {
-					case 'fast':
-					case 5:
-						newInterval = 5000;
-						break;
-					case 15:
-						newInterval = 15000;
-						break;
-					case 30:
-						newInterval = 30000;
-						break;
-					case 60:
-						newInterval = 60000;
-						break;
-					case 120:
-						newInterval = 120000;
-						break;
-					case 'long-polling':
-						// Allow long polling (experimental).
-						settings.mainInterval = 0;
-						return 0;
-					default:
+				if ( 'fast' === speed ) {
+					// Special case, see below.
+					newInterval = 5000;
+				} else if ( 'long-polling' === speed ) {
+					// Allow long polling (experimental).
+					settings.mainInterval = 0;
+					return 0;
+				} else {
+					speed = parseInt( speed, 10 );
+
+					if ( speed >= 1 && speed <= 3600 ) {
+						newInterval = speed * 1000;
+					} else {
 						newInterval = settings.originalInterval;
+					}
 				}
 
 				if ( settings.minimalInterval && newInterval < settings.minimalInterval ) {
 					newInterval = settings.minimalInterval;
 				}
 
+				// Special case, runs for a number of ticks then reverts to the previous interval.
 				if ( 5000 === newInterval ) {
```

Comment:
Now the interval can be specified as any number, should this check for `fast` or lower? 

```suggestion
				if ( newInterval <= 5000 ) {
```

Reply:
The idea was for all limitations to be removed. If a plugin wants to set this to do AJAX requests every second, thinking it would know what this means for the server and should be allowed to. This can be reset at any time to any value, including from the (PHP) filter for the AJAX response. Example use case may be to try to do simultaneous editing where every keystroke is sent to the server. Think Google docs (used to) work like that.

Also, the same countdown of hearbeat ticks can be controlled from a plugin. Perhaps better to leave it to plugins?

Reply:
Got it, the thought behind the question was that it seems odd this will be hit for `heartbeat.interval( 'fast' )` and `heartbeat.interval( 5 )` but not `4.9`. But having `ticks` as a special case is weird anyway so I'm not invested :)

Reply:
> having ticks as a special case is weird anyway

Yea, this was an exception but frankly can't remember why it work this way. Seems the countdown should have been added to the other script that needs it.